s3/src/main/scala/akka/stream/alpakka/s3/javadsl/S3Client.scala

```diff
@@ -195,10 +223,14 @@ final class S3Client(s3Settings: S3Settings, system: ActorSystem, mat: Materiali
    */
   def request(bucket: String,
               key: String,
+              versionId: String,
```

Comment:
Wouldn't it be better to go with Optional instead of a nullable String?
We might deprecate the `request` without the parameter.